### ISSUE TITLE
URL crashes the AI Chatbot and the whole page

### ISSUE THREAD

**Issue by mobiware** [sendbird/chat-ai-widget#314]
Since we've upgraded to version `1.7.3` of the SDK, we're running into an issue that we didn't experience with version `1.6.1`

We have a function call that returns a URL, and this URL is included by the chatbot into its response to the user as follows (this is only an excerpt from the full message showing the Markdown-formatted URL)
```
Your renewal lease has been fully executed. You can find a copy of the renewal lease [here](https://res.cloudinary.com/darwin-homes/image/authenticated/v1/sandbox/Ewmtbp/7781%20Kbujv%20Il/0021-55-88/xlsbsxx-eqpcuwu.pdf?__cld_token__=exp=1721913788~hmac=2620cdef709694691711b33a68f7cf41465d12f8343c54f58a09167f479ea2e1).
```

As soon as the chatbot displays this part of the message, it crashes and the whole page with it (whole page gets blank).

The exception that can be seen in the console is:
```
Uncaught (in promise) URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at D0 (chunk-EPLYPIMA.js?v=63f1ceef:27766:11)
    at replacer (chunk-EPLYPIMA.js?v=63f1ceef:28817:538)
    at chunk-EPLYPIMA.js?v=63f1ceef:28806:59
    at Array.forEach (<anonymous>)
    at chunk-EPLYPIMA.js?v=63f1ceef:28805:20
    at chunk-EPLYPIMA.js?v=63f1ceef:28809:11
    at Array.forEach (<anonymous>)
    at chunk-EPLYPIMA.js?v=63f1ceef:28797:7
    at Array.forEach (<anonymous>)
```

**Comment by mobiware** [comment]
So we've debugged a little bit and we found out that the URL that gets passed to `decodeURIComponent` is:
`https://res.cloudinary.com/darwin-homes/image/authenticated/v1/sandbox/Ewmtbp/7781%`

which is not the original URL (it's truncated). The fact that the URL is ending with `%` is what makes it malformed.

Looking at the `D0` function that calls `decodeURIComponent`, it looks like it corresponds to the new `asSafeURL` function
<img width="601" alt="Capture d’écran 2024-07-25 à 14 55 30" src="https://github.com/user-attachments/assets/d29eb435-6ace-46a6-864f-ce3db9e2814d">

It also seems like this function is being called as part of some Regexp manipulation of the URL, so perhaps there's a bug in the Regexp ?
<img width="1728" alt="Capture d’écran 2024-07-25 à 14 51 14" src="https://github.com/user-attachments/assets/7d7a1322-63fc-4484-b8b7-3c87dd06f8d8">

**Comment by bang9** [comment]
Thanks for letting me know. I'll fix this.

**Comment by mobiware** [comment]
@bang9 any ETA for a new version of the SDK including this fix ?

**Comment by bang9** [comment]
Hello @mobiware It has been released in v1.7.4 now!

**Comment by mobiware** [comment]
@bang9 awesome, thanks!
